R/createCompoundDbPackage.R

```diff
@@ -0,0 +1,149 @@
+#' @title Create a CompoundDb database
+#'
+#' @description
+#' 
+#' `createCompoundDb` creates a `SQLite`-based `CompoundDb` object/database
+#' from a compound resource provided as a `data.frame` or `tbl`.
+#' An additional `data.frame` providing metadata information is mandatory.
+#' Required columns for the `data.frame` providing the compound information are:
+#'
+#' 
+#' The metadata `data.frame` is supposed to have two columns named `"name"` and
+#' `"value"` providing the following minimal information as key-value pairs:
+#' + `"source"`: the source from which the data was retrieved (e.g. `"HMDB"`).
+#' + `"url"`: the url from which the original data was retrieved.
+#' + `"source_version"`: the version from the original data source
+#'   (e.g. `"v4"`).
+#' + `"source_date"`: the date when the original data source was generated.
+#' + `"organism"`: the organism. Should be in the form `"Hsapiens"` or
+#'   `"Mmusculus"`.
+#'
+#' @details
+#'
+#' Metadata information is also used to create the file name for the database
+#' file. The name starts with `"CompoundDb"`, followed by the organism, the
+#' data source and its version. A compound database file for HMDB version 4
+#' with human metabolites will thus be named: `"CompoundDb.Hsapiens.HMDB.v4"`.
+#' 
+#' @param x `data.frame` or `tbl` with the compound annotations. See
+#'     description for details.
+#'
+#' @param metadata `data.frame` with metadata information. See description for
+#'     details.
+#'
+#' @param path `character(1)` with the path to the directory where the database
+#'     file should be written. Defaults to the current directory.
+#'
+#' @return A `character` with the database name (invisibly).
+#' 
+#' @importFrom DBI dbDriver dbWriteTable dbExecute
+#' @importFrom RSQLite dbConnect
+#'
+#' @export
+#'
+#' @author Johannes Rainer
+#'
+#' @md
+#'
+#' @examples
+#'
+#' ## Read compounds for a HMDB subset
+#' fl <- system.file("extdata/hmdb/hmdb_sub.xml", package = "PeakABro")
+#' cmps <- generate_hmdb_tbl(fl)
+#'
+#' ## Create a metadata data.frame for the compounds.
+#' metad <- data.frame(name = c("source", "url", "source_version",
```

Comment:
data_frame to create tibble? (always drop=FALSE and no stringasfactor nonsense)

Reply:
Makes sense - I'm still hesitant to switch completely to the tidyverse :)